src/main/scala/com/github/zenpie/macrowave/internal/scanner/RuleParser.scala

```diff
@@ -45,6 +52,13 @@ trait RuleParser extends MacroUtils {
         tokenDefinition(tree, name, tpt, value)
     }
 
+    if (whiteSpaces.isEmpty || whiteSpaces.size == 1) {
+      grammar.whiteSpace = whiteSpaces.headOption
+    } else {
+      val whiteSpaceRuleNames = whiteSpaces.toList.map(grammar.terminalNames).sorted
+      c.error(c.enclosingPosition, s"Multiple definitions of whiteSpace-token: ${whiteSpaceRuleNames.mkString(", ")}!")
```

Comment:
Instead of `c.enclosingPosition` we could report the error at each white space rule which is not the first one. The error message could be "white space rule is already defined".